src/core/sr.ml

```diff
@@ -131,13 +131,14 @@ let symb_to_tenv : pre_rule Pos.loc -> sym list -> index_tbl -> term -> tbox =
   in
   symb_to_tenv t
 
-(** [check_rule bmap r] checks whether the pre-rule [r] is well-typed and then
-    construct the corresponding rule. The “bultin map” [bmap] is passed to the
-    unification engine. Note that [Fatal] is raised in case of error. *)
-let check_rule : sym StrMap.t -> pre_rule Pos.loc -> rule = fun bmap pr ->
+(** [check_rule r] checks whether the pre-rule [r] is well-typed in
+   signature state [ss] and then construct the corresponding rule. Note that
+   [Fatal] is raised in case of error. *)
+let check_rule : Scope.pre_rule Pos.loc -> rule = fun pr ->
   (* Unwrap the contents of the pre-rule. *)
-  let (pos, (s, h), lhs, vars, rhs_vars, arities) =
-    let Pos.{elt={pr_sym;pr_lhs;pr_vars;pr_rhs;pr_arities;_}; pos} = pr in
+  let (pos, s, lhs, vars, rhs_vars, arities) =
+    let Pos.{elt=Scope.{pr_sym;pr_lhs;pr_vars;pr_rhs;pr_arities;_}; pos} = pr
+    in
```

Comment:
Please, no, that is horrible style. Rather start the function with `let open Scope in`. Or reinstate the `open Scope`, which was not a problem at all.